ocaml/xapi/helpers.ml

```diff
@@ -797,6 +797,14 @@ let is_pool_master ~__context ~host =
   let master_id = Db.Host.get_uuid ~__context ~self:master in
   host_id = master_id
 
+let assert_we_are_master ~__context =
+  let host = get_localhost ~__context in
+  if not (is_pool_master ~__context ~host) then
+    raise
+      Api_errors.(
+        Server_error (host_is_slave, [Db.Host.get_address ~__context ~self:host])
```

Comment:
The parameter to HOST_IS_SLAVE must be the coordinator's address, not the local address (as done in the original function in xapi_psr.ml).